x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/index/engine/FollowingEngine.java

```diff
@@ -58,8 +58,12 @@ private void preFlight(final Operation operation) {
         final long maxSeqNoOfUpdatesOrDeletes = getMaxSeqNoOfUpdatesOrDeletes();
         assert maxSeqNoOfUpdatesOrDeletes != SequenceNumbers.UNASSIGNED_SEQ_NO : "max_seq_no_of_updates is not initialized";
         if (hasBeenProcessedBefore(index)) {
-            return IndexingStrategy.processButSkipLucene(false, index.seqNo(), index.version());
-
+            if (index.origin() == Operation.Origin.PRIMARY) {
+                final AlreadyProcessedFollowingEngineException error = new AlreadyProcessedFollowingEngineException(shardId, index.seqNo());
```

Comment:
can you add a comment as to why we do this here (and how it's dealt with from the outside)?

Reply:
I added a comment.